diff --git a/chrome/browser/resources/cyfs_init/cyfs_init.js b/chrome/browser/resources/cyfs_init/cyfs_init.js
index ccf0b55595..3fc3f8f605 100644
--- a/chrome/browser/resources/cyfs_init/cyfs_init.js
+++ b/chrome/browser/resources/cyfs_init/cyfs_init.js
@@ -12,7 +12,7 @@ function returnRuntimePorcessStatus(status) {
         // window.location.href = "cyfs://static/browser.html";
         let browser_url = loadTimeData.getStringF('browser_url');
         if (isFirstRun) {
-            browser_url = browser_url  '?first=true';
+            browser_url = browser_url + '?first=true';
         }
         window.location.href = browser_url;
         if (UpdateIntervalID)
