diff --git a/chrome/app/chromium_strings.grd b/chrome/app/chromium_strings.grd
index c3079d0eb8..d8f79d505a 100644
--- a/chrome/app/chromium_strings.grd
+++ b/chrome/app/chromium_strings.grd
@@ -277,10 +277,10 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
       </if>
 
       <message name="IDS_PRODUCT_NAME" desc="The Chrome application name">
-        CYFS Browser
+        Cyfs Browser
       </message>
       <message name="IDS_SHORT_PRODUCT_NAME" desc="The Chrome application short name.">
-        CYFS Browser
+        Cyfs Browser
       </message>
       <if expr="is_win">
         <message name="IDS_SXS_SHORTCUT_NAME" desc="Unused in Chromium builds" translateable="false">
@@ -447,7 +447,7 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
         </message>
       </if>
       <message name="IDS_FR_CUSTOMIZE_DEFAULT_BROWSER" desc="Default browser checkbox label">
-        Make Chromium the default browser
+        Make Cyfs Browser the default browser
       </message>
       <if expr="use_titlecase">
         <message name="IDS_STATUS_TRAY_KEEP_CHROME_RUNNING_IN_BACKGROUND" desc="In Title Case: The checkbox in the status tray context menu that controls whether chrome keeps running in the background after the last window is closed">
@@ -767,7 +767,7 @@ Permissions you've already given to websites and apps may apply to this account.
 
       <if expr="use_titlecase and not chromeos_ash">
         <message name="IDS_ABOUT" desc="In Title Case: The text label of the About Chrome menu item">
-          About &amp;Chromium
+          About &amp;Cyfs Browser
         </message>
         <message name="IDS_RELAUNCH_TO_UPDATE" desc="In Title Case: The text label of the relaunch to update Chrome menu item">
           Relaunch to Update &amp;Chromium
diff --git a/chrome/app/resources/chromium_strings_en-GB.xtb b/chrome/app/resources/chromium_strings_en-GB.xtb
index c16ba306f1..2bcf63df86 100644
--- a/chrome/app/resources/chromium_strings_en-GB.xtb
+++ b/chrome/app/resources/chromium_strings_en-GB.xtb
@@ -6,11 +6,11 @@
 <translation id="1065672644894730302">Your preferences cannot be read.
 
 Some features may be unavailable and changes to preferences won't be saved.</translation>
-<translation id="1098170124587656448">Chromium can help keep you safe from data breaches, bad extensions and more</translation>
+<translation id="7215125898567943664">Cyfs Browser can help keep you safe from data breaches, bad extensions and more</translation>
 <translation id="1104942323762546749">Chromium wants to export your passwords. Type your Windows password to allow this.</translation>
-<translation id="113122355610423240">Chromium is your default browser</translation>
+<translation id="8713042369605541296">Cyfs Browser is your default browser</translation>
 <translation id="1131805035311359397">To check if your passwords are safe from data breaches and other security issues, <ph name="BEGIN_LINK" />sign in to Chromium<ph name="END_LINK" />.</translation>
-<translation id="1185134272377778587">About Chromium</translation>
+<translation id="1613212111517732709">About Cyfs Browser</translation>
 <translation id="1203500561924088507">Thanks for installing. You must restart your browser before using <ph name="BUNDLE_NAME" />.</translation>
 <translation id="1315551408014407711">Set up your new Chromium profile</translation>
 <translation id="1396446129537741364">Chromium is trying to show passwords.</translation>
@@ -38,7 +38,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 
 Permissions that you've already given to websites and apps may apply to this account. You can manage your Google Accounts in <ph name="SETTINGS_LINK_BEGIN" />Settings<ph name="SETTINGS_LINK_END" />.</translation>
 <translation id="1953553007165777902">Downloading… <ph name="MINUTE" /> minute(s) remaining</translation>
-<translation id="1966382378801805537">Chromium cannot determine or set the default browser</translation>
+<translation id="8533470113944906162">Cyfs Browser cannot determine or set the default browser</translation>
 <translation id="2008474315282236005">This will delete 1 item from this device. To retrieve your data later, sign in to Chromium as <ph name="USER_EMAIL" />.</translation>
 <translation id="2018879682492276940">Installation failed. Please try again.</translation>
 <translation id="2020032459870799438">To check if your other passwords are safe from data breaches and other security issues, <ph name="BEGIN_LINK" />sign in to Chromium<ph name="END_LINK" />.</translation>
@@ -69,7 +69,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="2847479871509788944">Remove from Chromium...</translation>
 <translation id="2850691299438350830">Keeps you safe on Chromium and may be used to improve your security in other Google apps when you are signed in</translation>
 <translation id="2885378588091291677">Task Manager</translation>
-<translation id="8634147177990539032">About &amp;CYFS Browser</translation>
+<translation id="6886160333384257561">About &amp;Cyfs Browser</translation>
 <translation id="2915996080311180594">Restart later</translation>
 <translation id="2928420929544864228">Installation complete.</translation>
 <translation id="2977470724722393594">Chromium is up to date</translation>
@@ -93,7 +93,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="3283186697780795848">Chromium version <ph name="PRODUCT_VERSION" /> is installed</translation>
 <translation id="3286538390144397061">Restart Now</translation>
 <translation id="328888136576916638">Google API keys are missing. Some functionality of Chromium will be disabled.</translation>
-<translation id="3296368748942286671">Continue running background apps when Chromium is closed</translation>
+<translation id="7665642693549815118">Continue running background apps when Cyfs Browser is closed</translation>
 <translation id="3349211344363550267">Chromium recommends that you don't download or open this file</translation>
 <translation id="3350761136195634146">A Chromium profile with this account already exists</translation>
 <translation id="3387527074123400161">Chromium OS</translation>
@@ -134,7 +134,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="4216212958613226427">This language is used to display the Chromium UI</translation>
 <translation id="4222932583846282852">Cancelling...</translation>
 <translation id="4230135487732243613">Link my Chromium data to this account</translation>
-<translation id="4271805377592243930">Get help with Chromium</translation>
+<translation id="5568058375997949195">Get help with Cyfs Browser</translation>
 <translation id="4285930937574705105">Installation failed due to unspecified error. If Google Chrome is currently running, please close it and try again.</translation>
 <translation id="4304713468139749426">Password Manager</translation>
 <translation id="4334294535648607276">Download complete.</translation>
@@ -170,7 +170,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="5114678101347489141">Chromium is exploring new features that allow sites to deliver the same browsing experience using less of your info</translation>
 <translation id="5224391634244552924">No saved passwords. Chromium can check your passwords when you save them.</translation>
 <translation id="5252179775517634216"><ph name="EXISTING_USER" /> is already signed in to this Chromium profile. This will create a new Chromium profile for <ph name="USER_EMAIL_ADDRESS" /></translation>
-<translation id="5277894862589591112">To apply your changes, relaunch Chromium</translation>
+<translation id="4585823663498373655">To apply your changes, relaunch Chromium</translation>
 <translation id="5296845517486664001">When trials are on and if Chromium has randomly placed you in an active trial, your browsing history affects the ads that you see and the interests as estimated below. To protect your privacy, Chromium deletes your interests on a rolling basis each month.</translation>
 <translation id="5358375970380395591">You are signing in with a managed account and giving its administrator control over your Chromium profile. Your Chromium data, such as your apps, bookmarks, history, passwords and other settings will become permanently tied to <ph name="USER_NAME" />. You will be able to delete this data via the Google Accounts Dashboard, but you will not be able to associate this data with another account. <ph name="LEARN_MORE" /></translation>
 <translation id="5386450000063123300">Updating Chromium (<ph name="PROGRESS_PERCENT" />)</translation>
@@ -183,7 +183,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="5487574057737591516">To protect your privacy, we auto-delete your interests that are older than four weeks. As you keep browsing, an interest might appear on the list again. And if Chromium gets it wrong or you don't want to see certain ads, you can remove an interest.</translation>
 <translation id="549669000822060376">Please wait while Chromium installs the latest system updates.</translation>
 <translation id="5496810170689441661">Chromium is trying to edit passwords. Type your Windows password to allow this.</translation>
-<translation id="5527463683072221100">Open PDFs in Chromium</translation>
+<translation id="7442489743707017612">Open PDFs in Chromium</translation>
 <translation id="5623402015214259806">{0,plural, =0{A Chromium update is available}=1{A Chromium update is available}other{A Chromium update has been available for # days}}</translation>
 <translation id="5634636535844844681">Chromium requires Windows 7 or higher.</translation>
 <translation id="5653831366781983928">Please restart Chromium now</translation>
@@ -214,11 +214,11 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="6219195342503754812">{0,plural, =0{Chromium will relaunch now}=1{Chromium will relaunch in 1 second}other{Chromium will relaunch in # seconds}}</translation>
 <translation id="6241367896540709610">Chromium needs storage access permission to download files</translation>
 <translation id="6245734527075554892">Checks URLs with a list of unsafe sites stored in Chromium</translation>
-<translation id="6248213926982192922">Make Chromium the default browser</translation>
+<translation id="3666327966893632721">Make Cyfs Browser the default browser</translation>
 <translation id="6266342355635466082">Chromium can't check for updates. Try checking your Internet connection.</translation>
 <translation id="6268381023930128611">Sign out of Chromium?</translation>
 <translation id="6273793429163604305">On your marks…</translation>
-<translation id="6281746429495226318">Customise your Chromium profile</translation>
+<translation id="2828234732082468538">Customise your Cyfs Browser profile</translation>
 <translation id="6290827346642914212">Name your Chromium profile</translation>
 <translation id="6295779123002464101"><ph name="FILE_NAME" /> may be dangerous, so Chromium has blocked it.</translation>
 <translation id="6309712487085796862">Chromium is using your camera.</translation>
@@ -273,11 +273,11 @@ Some features may be unavailable. Please specify a different profile directory o
 <translation id="7246575524853130370">Your interests as estimated by Chromium</translation>
 <translation id="7265671031637041256">Chromium blocked this file because it's encrypted. Ask its owner to decrypt it</translation>
 <translation id="7309928523159922338">Ad measurement allows sites that you visit to request information from Chromium that helps the site measure the performance of their ads. Ad measurement restricts cross-site tracking by transferring as little information as possible between sites.</translation>
-<translation id="731795002583552498">Updating Chromium</translation>
+<translation id="7840644711321880954">Updating Cyfs Browser</translation>
 <translation id="7318036098707714271">Your preferences file is corrupt or invalid.
 
 Google Chrome is unable to recover your settings.</translation>
-<translation id="5969877561795249315">Cyfs Browser</translation>
+<translation id="138464974652861409">Cyfs Browser</translation>
 <translation id="7339898014177206373">New window</translation>
 <translation id="734373864078049451">Your web, bookmarks and other Chromium stuff live here.</translation>
 <translation id="7349591376906416160">Your system administrator has configured Chromium to open <ph name="ALTERNATIVE_BROWSER_NAME" /> to access <ph name="TARGET_URL_HOSTNAME" />.</translation>
@@ -309,7 +309,7 @@ Google Chrome is unable to recover your settings.</translation>
 <translation id="7877292582355102282"><ph name="BEGIN_BOLD" />How you can manage your data:<ph name="END_BOLD" /> To protect your privacy, we auto-delete your interests that are older than four weeks. As you keep browsing, an interest might appear on the list again. Or you can remove interests that you don’t want Chromium to consider.</translation>
 <translation id="788352880609666695">The following accounts are available for this Chromium profile</translation>
 <translation id="7937630085815544518">You were signed in to Chromium as <ph name="USER_EMAIL_ADDRESS" />. Please use the same account to sign in again.</translation>
-<translation id="7975919845073681630">This is a secondary installation of Chromium, and cannot be made your default browser.</translation>
+<translation id="8776078950548090698">This is a secondary installation of Cyfs Browser, and cannot be made your default browser.</translation>
 <translation id="7982481011030453202">If a setting doesn't show on this page, look in your <ph name="LINK_BEGIN" />
       Chromium OS settings<ph name="LINK_END" /></translation>
 <translation id="7997934263947464652">Extensions, apps and themes from unknown sources can harm your device. Chromium recommends only installing them from the <ph name="IDS_EXTENSION_WEB_STORE_TITLE" /></translation>
@@ -334,7 +334,7 @@ Google Chrome is unable to recover your settings.</translation>
 <translation id="8558383651099478961">Your parent has turned off 'Permissions for sites, apps and extensions' for Chromium. Adding this <ph name="EXTENSION_TYPE_PARAMETER" /> is not allowed.</translation>
 <translation id="8561285221328166383">To get future Chromium updates, you'll need macOS 10.13 or later. This computer is using macOS 10.12.</translation>
 <translation id="8568283329061645092">Chromium can check your passwords when you sign in with your Google Account</translation>
-<translation id="6298955232712833087">Copyright <ph name="YEAR" /> The Cyfs Browser Authors. All rights reserved.</translation>
+<translation id="7267362170917110807">Copyright <ph name="YEAR" /> The Cyfs Browser Authors. All rights reserved.</translation>
 <translation id="8608079656141766906"><ph name="BEGIN_BOLD" />How we use this data:<ph name="END_BOLD" /> Sites can store information with Chromium about your interests. For example, if you visit a site to buy shoes for a marathon, the site might define your interest as running marathons. Later, if you visit a different site to register for a race, that site can show you an ad for running shoes based on your interests.</translation>
 <translation id="8619360774459241877">Launching Chromium...</translation>
 <translation id="8621669128220841554">Installation failed due to unspecified error. Please download Chromium again.</translation>
diff --git a/chrome/app/resources/chromium_strings_zh-CN.xtb b/chrome/app/resources/chromium_strings_zh-CN.xtb
index 8a38608a83..1d5f1c3081 100644
--- a/chrome/app/resources/chromium_strings_zh-CN.xtb
+++ b/chrome/app/resources/chromium_strings_zh-CN.xtb
@@ -4,11 +4,11 @@
 <translation id="1026101648481255140">继续安装</translation>
 <translation id="1029669172902658969">重新启动以更新 Chromium 操作系统(&amp;C)</translation>
 <translation id="1065672644894730302">系统无法读取您的偏好设置。某些功能可能无法使用，并且对偏好设置所做的更改不会保存。</translation>
-<translation id="1098170124587656448">Chromium 有助于保护您免受数据泄露、不良扩展程序等问题的影响</translation>
+<translation id="7215125898567943664">Cyfs Browser 有助于保护您免受数据泄露、不良扩展程序等问题的影响</translation>
 <translation id="1104942323762546749">Chromium 想导出您的密码。请输入您的 Windows 密码以允许此操作。</translation>
-<translation id="113122355610423240">Chromium 是您的默认浏览器</translation>
+<translation id="8713042369605541296">Cyfs Browser 是您的默认浏览器</translation>
 <translation id="1131805035311359397">若要检查您的密码是否未曾遭遇数据泄露和其他安全问题，请<ph name="BEGIN_LINK" />登录 Chromium<ph name="END_LINK" />。</translation>
-<translation id="1185134272377778587">关于 Chromium</translation>
+<translation id="1613212111517732709">关于 Cyfs Browser</translation>
 <translation id="1203500561924088507">感谢您完成安装。您必须重启浏览器才能使用 <ph name="BUNDLE_NAME" />。</translation>
 <translation id="1315551408014407711">设置您的新 Chromium 个人资料</translation>
 <translation id="1396446129537741364">Chromium 正尝试显示密码。</translation>
@@ -36,7 +36,7 @@
 
 您已向网站和应用授予的权限可能会应用于此帐号。您可在<ph name="SETTINGS_LINK_BEGIN" />设置<ph name="SETTINGS_LINK_END" />部分中管理您的 Google 帐号。</translation>
 <translation id="1953553007165777902">正在下载…还需 <ph name="MINUTE" /> 分钟</translation>
-<translation id="1966382378801805537">Chromium 无法确定或设置默认浏览器</translation>
+<translation id="8533470113944906162">Cyfs Browser 无法确定或设置默认浏览器</translation>
 <translation id="2008474315282236005">这会从这台设备上删除 1 项内容。如果日后想要检索您的数据，届时请以 <ph name="USER_EMAIL" /> 的身份登录 Chromium。</translation>
 <translation id="2018879682492276940">安装失败。请重试。</translation>
 <translation id="2020032459870799438">若要检查您的其他密码是否未曾遭遇数据泄露和其他安全问题，请<ph name="BEGIN_LINK" />登录 Chromium<ph name="END_LINK" />。</translation>
@@ -67,7 +67,7 @@
 <translation id="2847479871509788944">从Chromium中移除…</translation>
 <translation id="2850691299438350830">保护您在 Chromium 中免受安全威胁，登录状态下还可让您更安全地使用其他 Google 应用</translation>
 <translation id="2885378588091291677">任务管理器</translation>
-<translation id="8634147177990539032">关于 Cyfs Browser(&amp;C)</translation>
+<translation id="6886160333384257561">关于 Cyfs Browser(&amp;C)</translation>
 <translation id="2915996080311180594">暂不重启</translation>
 <translation id="2928420929544864228">安装完毕。</translation>
 <translation id="2977470724722393594">Chromium 已是最新版本</translation>
@@ -91,7 +91,7 @@
 <translation id="3283186697780795848">已安装 Chromium 版本 <ph name="PRODUCT_VERSION" /></translation>
 <translation id="3286538390144397061">立即重新启动</translation>
 <translation id="328888136576916638">缺少 Google API 密钥，因此 Chromium 的部分功能将无法使用。</translation>
-<translation id="3296368748942286671">关闭 Chromium 后继续运行后台应用</translation>
+<translation id="7665642693549815118">关闭 Cyfs Browser 后继续运行后台应用</translation>
 <translation id="3349211344363550267">Chromium 建议您不要下载或打开此文件</translation>
 <translation id="3350761136195634146">已有一份使用此帐号创建的 Chromium 个人资料</translation>
 <translation id="3387527074123400161">Chromium 操作系统</translation>
@@ -132,7 +132,7 @@
 <translation id="4216212958613226427">使用此语言显示 Chromium 界面</translation>
 <translation id="4222932583846282852">正在取消…</translation>
 <translation id="4230135487732243613">是否将您的 Chromium 数据与此帐号相关联？</translation>
-<translation id="4271805377592243930">获取有关 Chromium 的帮助</translation>
+<translation id="5568058375997949195">获取有关 Cyfs Browser 的帮助</translation>
 <translation id="4285930937574705105">未知错误导致安装失败。如果 Chromium 正在运行，请将其关闭，然后重试。</translation>
 <translation id="4304713468139749426">密码管理器</translation>
 <translation id="4334294535648607276">下载完毕。</translation>
@@ -168,7 +168,7 @@
 <translation id="5114678101347489141">Chromium 正在探索各种新功能，以便让网站既能减少使用您的信息，又能提供同样出色的浏览体验</translation>
 <translation id="5224391634244552924">尚未保存任何密码。您需要先保存密码才能使用 Chromium 的密码检查功能。</translation>
 <translation id="5252179775517634216"><ph name="EXISTING_USER" />已登录此 Chromium 个人资料。这样做会为 <ph name="USER_EMAIL_ADDRESS" /> 创建一份新的 Chromium 个人资料</translation>
-<translation id="5277894862589591112">要想应用您的更改，请重新启动 Chromium</translation>
+<translation id="4585823663498373655">要想应用您的更改，请重新启动 Cyfs Browser</translation>
 <translation id="5296845517486664001">在试用期间，如果 Chromium 随机将您加入到了其中某项正在进行的试用活动，您的浏览记录会影响您看到的广告以及系统推测出的兴趣（如下所示）。为了保护您的隐私，Chromium 每个月都会按保留时间先后顺序删除您的部分兴趣。</translation>
 <translation id="5358375970380395591">您目前登录的帐号是一个受管理帐号，该帐号的管理员将能够控制您的 Chromium 个人资料。您的 Chromium 数据（例如您的应用、书签、历史记录、密码和其他设置）将永远与 <ph name="USER_NAME" /> 相关联。您可以通过 Google 帐号信息中心删除这些数据，但无法将这些数据与其他帐号相关联。<ph name="LEARN_MORE" /></translation>
 <translation id="5386450000063123300">正在更新 Chromium (<ph name="PROGRESS_PERCENT" />)</translation>
@@ -180,7 +180,7 @@
 <translation id="5487574057737591516">为了保护您的隐私，如果您的兴趣保留时间超过 4 周，系统便会自动将其删除。随着您继续浏览，某项兴趣可能会再次出现在列表中。如果 Chromium 的推测不准确，或者您不想看到某些广告，可以移除某项兴趣。</translation>
 <translation id="549669000822060376">Chromium正在安装最新的系统更新，请稍候。</translation>
 <translation id="5496810170689441661">Chromium 正在尝试修改密码。请输入您的 Windows 密码以允许此操作。</translation>
-<translation id="5527463683072221100">在 Chromium 中打开 PDF 文件</translation>
+<translation id="7442489743707017612">在 Cyfs Browser 中打开 PDF 文件</translation>
 <translation id="5623402015214259806">{0,plural, =0{有一项可用的 Chromium 更新}=1{有一项可用的 Chromium 更新}other{有一项可用的 Chromium 更新（已发布 # 天）}}</translation>
 <translation id="5634636535844844681">Chromium 仅支持 Windows 7 或更高版本的操作系统。</translation>
 <translation id="5653831366781983928">请立即重启 Chromium</translation>
@@ -211,11 +211,11 @@
 <translation id="6219195342503754812">{0,plural, =0{Chromium 将立即重新启动}=1{Chromium 将在 1 秒后重新启动}other{Chromium 将在 # 秒后重新启动}}</translation>
 <translation id="6241367896540709610">Chromium 需要获得存储空间使用权限才能下载文件</translation>
 <translation id="6245734527075554892">将网址与存储在 Chromium 中的不安全网站列表进行比对</translation>
-<translation id="6248213926982192922">将 Chromium 设置为默认浏览器</translation>
+<translation id="3666327966893632721">将 Cyfs Browser 设置为默认浏览器</translation>
 <translation id="6266342355635466082">Chromium 无法检查是否有更新。请检查您的互联网连接。</translation>
 <translation id="6268381023930128611">退出 Chromium？</translation>
 <translation id="6273793429163604305">即将就绪…</translation>
-<translation id="6281746429495226318">自定义您的 Chromium 个人资料</translation>
+<translation id="2828234732082468538">自定义您的 Cyfs Browser 个人资料</translation>
 <translation id="6290827346642914212">请为您的 Chromium 个人资料命名</translation>
 <translation id="6295779123002464101"><ph name="FILE_NAME" /> 可能存在危险，因此 Chromium 已将其拦截。</translation>
 <translation id="6309712487085796862">Chromium 正在使用您的摄像头。</translation>
@@ -270,11 +270,11 @@
 <translation id="7246575524853130370">Chromium 推测出的您的兴趣</translation>
 <translation id="7265671031637041256">Chromium 屏蔽了此文件，因为它已加密。请让文件所有者解密</translation>
 <translation id="7309928523159922338">借助 Ad measurement，您访问的网站可向 Chromium 请求信息，以便对网站上的广告进行效果衡量。Ad measurement 会在网站间传输尽可能少的信息，从而限制跨网站跟踪。</translation>
-<translation id="731795002583552498">正在更新 Chromium</translation>
+<translation id="7840644711321880954">正在更新 Cyfs Browser</translation>
 <translation id="7318036098707714271">您的偏好设置文件已损坏或无效。
 
 Chromium 无法恢复您的设置。</translation>
-<translation id="5969877561795249315">Cyfs Browser</translation>
+<translation id="138464974652861409">Cyfs Browser</translation>
 <translation id="7339898014177206373">打开新的窗口</translation>
 <translation id="734373864078049451">您的上网记录、书签和其他Chromium内容都保存在此处。</translation>
 <translation id="7349591376906416160">您的系统管理员已将 Chromium 配置为使用<ph name="ALTERNATIVE_BROWSER_NAME" />访问 <ph name="TARGET_URL_HOSTNAME" />。</translation>
@@ -306,7 +306,7 @@ Chromium 无法恢复您的设置。</translation>
 <translation id="7877292582355102282"><ph name="BEGIN_BOLD" />您可以如何管理自己的数据<ph name="END_BOLD" />：为了保护您的隐私，如果您的兴趣保留时间超过 4 周，系统便会自动将其删除。随着您继续浏览，某项兴趣可能会再次出现在列表中。您也可以移除不希望纳入 Chromium 考虑范围的兴趣。</translation>
 <translation id="788352880609666695">这份 Chromium 个人资料可以使用下列帐号</translation>
 <translation id="7937630085815544518">您之前登录 Chromium 时使用的帐号为 <ph name="USER_EMAIL_ADDRESS" />，请使用同一帐号再次登录。</translation>
-<translation id="7975919845073681630">这是 Chromium 的并行安装，您无法将此浏览器设为默认浏览器。</translation>
+<translation id="8776078950548090698">这是 Cyfs Browser 的并行安装，您无法将此浏览器设为默认浏览器。</translation>
 <translation id="7982481011030453202">如果此页面中未显示某项设置，请在 <ph name="LINK_BEGIN" />Chromium 操作系统设置<ph name="LINK_END" />中查找</translation>
 <translation id="7997934263947464652">来源不明的扩展程序、应用和主题可能会危害您的设备。Chromium 建议只从 <ph name="IDS_EXTENSION_WEB_STORE_TITLE" />安装这些内容</translation>
 <translation id="8013436988911883588">一旦 Chromium 获得了使用权限，网站便能请求您向其授予使用权限。</translation>
@@ -330,7 +330,7 @@ Chromium 无法恢复您的设置。</translation>
 <translation id="8558383651099478961">您的家长已为 Chromium 停用“网站、应用和扩展程序的权限”。因此，您无法添加该<ph name="EXTENSION_TYPE_PARAMETER" />。</translation>
 <translation id="8561285221328166383">若要接收后续 Chromium 更新，您需使用 macOS 10.13 或更高版本。该计算机目前使用的是 macOS 10.12。</translation>
 <translation id="8568283329061645092">您需要先登录 Google 帐号才能使用 Chromium 的密码检查功能。</translation>
-<translation id="6298955232712833087">版权所有<ph name="YEAR" /> The Cyfs Browser Authors。保留所有权利。</translation>
+<translation id="7267362170917110807">版权所有<ph name="YEAR" /> The Cyfs Browser Authors。保留所有权利。</translation>
 <translation id="8608079656141766906"><ph name="BEGIN_BOLD" />我们会如何使用这些数据<ph name="END_BOLD" />：网站可以将您的兴趣信息存储在 Chromium 中。例如，如果您访问某个网站购买马拉松跑鞋，系统可能会根据该网站推测您对跑马拉松感兴趣。之后，如果您访问其他网站报名参加比赛，相应网站就可以根据您的兴趣向您展示跑鞋广告。</translation>
 <translation id="8619360774459241877">正在启动 Chromium…</translation>
 <translation id="8621669128220841554">未指定的错误导致安装失败。请重新下载 Chromium。</translation>
diff --git a/chrome/app/resources/generated_resources_en-GB.xtb b/chrome/app/resources/generated_resources_en-GB.xtb
index f278c92817..3ac5176cce 100644
--- a/chrome/app/resources/generated_resources_en-GB.xtb
+++ b/chrome/app/resources/generated_resources_en-GB.xtb
@@ -2848,7 +2848,7 @@ and Ctrl+Alt+Brightness down to zoom out.</translation>
 <translation id="3714195043138862580">This demo device has been placed into a deprovisioned state.</translation>
 <translation id="3719826155360621982">Homepage</translation>
 <translation id="372062398998492895">CUPS</translation>
-<translation id="3721119614952978349">You and Google</translation>
+<translation id="3082888583566165225">Your Information</translation>
 <translation id="3722108462506185496">Error starting the virtual machine service. Please try again.</translation>
 <translation id="3727144509609414201">Available Wi-Fi networks</translation>
 <translation id="3727187387656390258">Inspect pop-up</translation>
diff --git a/chrome/app/resources/generated_resources_zh-CN.xtb b/chrome/app/resources/generated_resources_zh-CN.xtb
index 9ac03d7de0..b351771ac5 100644
--- a/chrome/app/resources/generated_resources_zh-CN.xtb
+++ b/chrome/app/resources/generated_resources_zh-CN.xtb
@@ -2822,7 +2822,7 @@
 <translation id="3714195043138862580">此演示设备目前处于“已取消配置”状态。</translation>
 <translation id="3719826155360621982">主页</translation>
 <translation id="372062398998492895">CUPS</translation>
-<translation id="3721119614952978349">您与 Google</translation>
+<translation id="3082888583566165225">您的信息</translation>
 <translation id="3722108462506185496">启动虚拟机服务时出错。请重试。</translation>
 <translation id="3727144509609414201">可用的 Wi-Fi 网络</translation>
 <translation id="3727187387656390258">审查弹出内容</translation>
diff --git a/chrome/app/settings_chromium_strings.grdp b/chrome/app/settings_chromium_strings.grdp
index 541432c04d..c2933f5691 100644
--- a/chrome/app/settings_chromium_strings.grdp
+++ b/chrome/app/settings_chromium_strings.grdp
@@ -9,14 +9,14 @@
   </if>
   <!-- About Page -->
   <message name="IDS_SETTINGS_ABOUT_PROGRAM" desc="Menu title for the About Chromium page.">
-    About Chromium
+    About Cyfs Browser
   </message>
   <message name="IDS_SETTINGS_GET_HELP_USING_CHROME" desc="Text of the button which takes the user to the Chrome help page.">
-    Get help with Chromium
+    Get help with Cyfs Browser
   </message>
   <if expr="not chromeos_ash">
     <message name="IDS_SETTINGS_UPGRADE_UPDATING" desc="Status label: Updating Chromium">
-      Updating Chromium
+      Updating Cyfs Browser
     </message>
     <message name="IDS_SETTINGS_UPGRADE_UPDATING_PERCENT" desc="Status label: Updating Chromium (90%)">
       Updating Chromium (<ph name="PROGRESS_PERCENT">$1<ex>90%</ex></ph>)
@@ -83,16 +83,16 @@
   <!-- Default Browser Page -->
   <if expr="not chromeos_ash">
     <message name="IDS_SETTINGS_DEFAULT_BROWSER_DEFAULT" desc="The text displayed when Chrome is the default browser">
-      Chromium is your default browser
+      Cyfs Browser is your default browser
     </message>
     <message name="IDS_SETTINGS_DEFAULT_BROWSER_MAKE_DEFAULT" desc="Default browser checkbox label">
-      Make Chromium the default browser
+      Make Cyfs Browser the default browser
     </message>
     <message name="IDS_SETTINGS_DEFAULT_BROWSER_ERROR" desc="The text displayed when Chrome cannot determine or set the default browser">
-      Chromium cannot determine or set the default browser
+      Cyfs Browser cannot determine or set the default browser
     </message>
     <message name="IDS_SETTINGS_DEFAULT_BROWSER_SECONDARY" desc="The text displayed when Chromium is installed in side-by-side mode, which does not support setting as the default browser.">
-      This is a secondary installation of Chromium, and cannot be made your default browser.
+      This is a secondary installation of Cyfs Browser, and cannot be made your default browser.
     </message>
   </if>
 
@@ -110,7 +110,7 @@
     To fix spelling errors, Chromium sends the text you type in text fields to Google
   </message>
   <message name="IDS_SETTINGS_RESTART_TO_APPLY_CHANGES" desc="Description in the snackbar to restart Chrome and apply changes.">
-    To apply your changes, relaunch Chromium
+    To apply your changes, relaunch Cyfs Browser
   </message>
 
   <if expr="not chromeos_ash and not chromeos_lacros">
@@ -124,7 +124,7 @@
 
   <!-- Site Settings -->
   <message name="IDS_SETTINGS_SITE_SETTINGS_PDFS_BLOCKED" desc="Label for the disabled option of the pdfs content setting.">
-    Open PDFs in Chromium
+    Open PDFs in Cyfs Browser
   </message>
 
   <!-- Privacy Guide -->
@@ -208,7 +208,7 @@
 
   <!-- Safety check -->
   <message name="IDS_SETTINGS_SAFETY_CHECK_PARENT_PRIMARY_LABEL_BEFORE" desc="This text describes what the safety check is. (It's an area of the Settings page where users can quickly check whether their safety-related settings are fully protecting them.)">
-    Chromium can help keep you safe from data breaches, bad extensions, and more
+    Cyfs Browser can help keep you safe from data breaches, bad extensions, and more
   </message>
   <message name="IDS_SETTINGS_SAFETY_CHECK_UPDATES_FAILED_OFFLINE" desc="This text describes that Chromium cannot check for updates because there currently is no internet connection.">
     Chromium can't check for updates. Try checking your internet connection.
@@ -246,7 +246,7 @@
   </message>
 <if expr="not chromeos_ash">
   <message name="IDS_SETTINGS_CUSTOMIZE_YOUR_CHROME_PROFILE" desc="Label of the link that takes you to the page to customize your chrome profile.">
-    Customize your Chromium profile
+    Customize your Cyfs Browser profile
   </message>
   <message name="IDS_SETTING_NAME_YOUR_PROFILE" desc="Title of the edit local profile name section on the manage profile page.">
     Name your Chromium profile
@@ -288,7 +288,7 @@
   <!-- System Page -->
   <if expr="not chromeos_ash">
     <message name="IDS_SETTINGS_SYSTEM_BACKGROUND_APPS_LABEL" desc="Label for the checkbox to allow apps to run in the background, when Chrome is closed.">
-      Continue running background apps when Chromium is closed
+      Continue running background apps when Cyfs Browser is closed
     </message>
   </if>
 
diff --git a/chrome/app/settings_strings.grdp b/chrome/app/settings_strings.grdp
index aaef2cea2f..018492a176 100644
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -3375,7 +3375,7 @@
 
   <!-- Sync / People Page -->
   <message name="IDS_SETTINGS_PEOPLE" desc="Name of the settings page which manages the user's relationship to Google.">
-    You and Google
+    Your Information
   </message>
   <message name="IDS_SETTINGS_CHANGE_PICTURE_PROFILE_PHOTO" desc="The text on the Google profile photo of the user.">
     Google Profile photo
diff --git a/chrome/browser/resources/settings/a11y_page/a11y_page.html b/chrome/browser/resources/settings/a11y_page/a11y_page.html
index 7b3696b735..49ff989777 100644
--- a/chrome/browser/resources/settings/a11y_page/a11y_page.html
+++ b/chrome/browser/resources/settings/a11y_page/a11y_page.html
@@ -51,10 +51,10 @@
             label="$i18n{accessibleImageLabelsTitle}"
             sub-label="$i18n{accessibleImageLabelsSubtitle}">
         </settings-toggle-button>
-        <cr-link-row class="hr" label="$i18n{moreFeaturesLink}"
+        <!-- <cr-link-row class="hr" label="$i18n{moreFeaturesLink}"
             on-click="onMoreFeaturesLinkClick_" sub-label="$i18n{a11yWebStore}"
             external>
-        </cr-link-row>
+        </cr-link-row> -->
       </div>
 <if expr="not chromeos_ash">
       <template is="dom-if" if="[[!captionSettingsOpensExternally_]]">
diff --git a/chrome/browser/resources/settings/about_page/about_page.html b/chrome/browser/resources/settings/about_page/about_page.html
index 28c284a581..f11989f2c6 100644
--- a/chrome/browser/resources/settings/about_page/about_page.html
+++ b/chrome/browser/resources/settings/about_page/about_page.html
@@ -131,8 +131,8 @@
         </div>
       </template>
 </if>
-      <cr-link-row class="hr" id="help" on-click="onHelpTap_"
-          label="$i18n{aboutGetHelpUsingChrome}" external></cr-link-row>
+      <!-- <cr-link-row class="hr" id="help" on-click="onHelpTap_"
+          label="$i18n{aboutGetHelpUsingChrome}" external></cr-link-row> -->
 <if expr="_google_chrome">
       <cr-link-row class="hr" id="reportIssue" on-click="onReportIssueTap_"
           hidden="[[!prefs.feedback_allowed.value]]"
diff --git a/chrome/browser/resources/settings/people_page/people_page.html b/chrome/browser/resources/settings/people_page/people_page.html
index 78504f333c..42d4eca34b 100644
--- a/chrome/browser/resources/settings/people_page/people_page.html
+++ b/chrome/browser/resources/settings/people_page/people_page.html
@@ -53,7 +53,7 @@
     <settings-animated-pages id="pages" section="people"
         focus-config="[[focusConfig_]]">
       <div route-path="default">
-        <template is="dom-if" if="[[shouldShowSyncAccountControl_(
+        <!-- <template is="dom-if" if="[[shouldShowSyncAccountControl_(
             syncStatus.syncSystemEnabled)]]">
           <settings-sync-account-control
               sync-status="[[syncStatus]]"
@@ -78,8 +78,6 @@
               </div>
               <div class="flex cr-row-gap cr-padded-text text-elide">
                 <span id="profile-name">[[profileName_]]</span>
-<!-- When the user is signed-in, the settings-sync-account-control is always
-shown on non-ChromeOS platforms -->
 <if expr="chromeos_ash">
                 <div class="secondary" hidden="[[!syncStatus.signedIn]]">
                   [[syncStatus.signedInUsername]]
@@ -102,21 +100,21 @@ shown on non-ChromeOS platforms -->
 </if>
             </template>
           </div>
-        </template> <!-- if="[[!shouldShowSyncAccountControl_()]]" -->
+        </template>  -->
 
-        <cr-link-row id="sync-setup"
+        <!-- <cr-link-row id="sync-setup"
             label="$i18n{syncAndNonPersonalizedServices}"
             sub-label="[[getSyncAndGoogleServicesSubtext_(syncStatus)]]"
             on-click="onSyncTap_"
             role-description="$i18n{subpageArrowRoleDescription}">
-        </cr-link-row>
+        </cr-link-row> -->
 
 <if expr="not chromeos_ash">
         <template is="dom-if" if="[[signinAllowed_]]">
-          <cr-link-row id="manage-google-account"
+          <!-- <cr-link-row id="manage-google-account"
               label="$i18n{manageGoogleAccount}"
               hidden="[[!shouldShowGoogleAccount_]]"
-              on-click="openGoogleAccount_" external></cr-link-row>
+              on-click="openGoogleAccount_" external></cr-link-row> -->
 
           <cr-link-row id="edit-profile"
               label="$i18n{profileNameAndPicture}"
diff --git a/chrome/browser/resources/settings/privacy_page/privacy_page.html b/chrome/browser/resources/settings/privacy_page/privacy_page.html
index 248f9d9a31..92403ff579 100644
--- a/chrome/browser/resources/settings/privacy_page/privacy_page.html
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page.html
@@ -86,7 +86,7 @@
             on-click="onPermissionsPageClick_"
             role-description="$i18n{subpageArrowRoleDescription}"></cr-link-row>
         <template is="dom-if" if="[[!isPrivacySandboxRestricted_]]">
-          <cr-link-row id="privacySandboxLinkRow"
+          <!-- <cr-link-row id="privacySandboxLinkRow"
               start-icon="settings20:experiment"
               class="hr"
               label="$i18n{privacySandboxTitle}"
@@ -95,7 +95,7 @@
               on-click="onPrivacySandboxClick_"
               external
               role-description="$i18n{subpageArrowRoleDescription}">
-          </cr-link-row>
+          </cr-link-row> -->
           <a id="privacySandboxLink" href="privacySandbox"
               target="_blank" tabindex="-1" aria-disabled="true"
               role="none"></a>
diff --git a/chrome/browser/ui/views/toolbar/toolbar_view.cc b/chrome/browser/ui/views/toolbar/toolbar_view.cc
index 55f36daf30..08534c9e7a 100644
--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -326,26 +326,26 @@ void ToolbarView::Init() {
   if (extensions_container)
     extensions_container_ = AddChildView(std::move(extensions_container));
 
-  if (base::FeatureList::IsEnabled(features::kChromeLabs)) {
-    chrome_labs_model_ = std::make_unique<ChromeLabsBubbleViewModel>();
-    UpdateChromeLabsNewBadgePrefs(browser_->profile(),
-                                  chrome_labs_model_.get());
-    if (ChromeLabsButton::ShouldShowButton(chrome_labs_model_.get(),
-                                           browser_->profile())) {
-      chrome_labs_button_ = AddChildView(std::make_unique<ChromeLabsButton>(
-          browser_view_, chrome_labs_model_.get()));
-
-      show_chrome_labs_button_.Init(
-          chrome_labs_prefs::kBrowserLabsEnabled,
-          browser_->profile()->GetPrefs(),
-          base::BindRepeating(&ToolbarView::OnChromeLabsPrefChanged,
-                              base::Unretained(this)));
-      // Set the visibility for the button based on initial enterprise policy
-      // value. Only call OnChromeLabsPrefChanged if there is a change from the
-      // initial value.
-      chrome_labs_button_->SetVisible(show_chrome_labs_button_.GetValue());
-    }
-  }
+  // if (base::FeatureList::IsEnabled(features::kChromeLabs)) {
+  //   chrome_labs_model_ = std::make_unique<ChromeLabsBubbleViewModel>();
+  //   UpdateChromeLabsNewBadgePrefs(browser_->profile(),
+  //                                 chrome_labs_model_.get());
+  //   if (ChromeLabsButton::ShouldShowButton(chrome_labs_model_.get(),
+  //                                          browser_->profile())) {
+  //     chrome_labs_button_ = AddChildView(std::make_unique<ChromeLabsButton>(
+  //         browser_view_, chrome_labs_model_.get()));
+
+  //     show_chrome_labs_button_.Init(
+  //         chrome_labs_prefs::kBrowserLabsEnabled,
+  //         browser_->profile()->GetPrefs(),
+  //         base::BindRepeating(&ToolbarView::OnChromeLabsPrefChanged,
+  //                             base::Unretained(this)));
+  //     // Set the visibility for the button based on initial enterprise policy
+  //     // value. Only call OnChromeLabsPrefChanged if there is a change from the
+  //     // initial value.
+  //     chrome_labs_button_->SetVisible(show_chrome_labs_button_.GetValue());
+  //   }
+  // }
 
   if (cast)
     cast_ = AddChildView(std::move(cast));
diff --git a/components/components_chromium_strings.grd b/components/components_chromium_strings.grd
index 4eeae4b8a2..12b47dc45a 100644
--- a/components/components_chromium_strings.grd
+++ b/components/components_chromium_strings.grd
@@ -286,7 +286,7 @@
       </message>
 
       <message name="IDS_VERSION_UI_LICENSE" desc="The label below the copyright message, containing the URLs.">
-        CYFS Browser is made possible by the <ph name="BEGIN_LINK_CHROMIUM">&lt;a target="_blank" href="$1"&gt;</ph>Chromium<ph name="END_LINK_CHROMIUM">&lt;/a&gt;</ph> open source project and other <ph name="BEGIN_LINK_OSS">&lt;a target="_blank" href="$2"&gt;</ph>open source software<ph name="END_LINK_OSS">&lt;/a&gt;</ph>.
+        Cyfs Browser is made possible by the <ph name="BEGIN_LINK_CHROMIUM">&lt;a target="_blank" href="$1"&gt;</ph>Chromium<ph name="END_LINK_CHROMIUM">&lt;/a&gt;</ph> open source project and other <ph name="BEGIN_LINK_OSS">&lt;a target="_blank" href="$2"&gt;</ph>open source software<ph name="END_LINK_OSS">&lt;/a&gt;</ph>.
       </message>
 
       <!-- Page Info -->
diff --git a/components/strings/components_chromium_strings_en-GB.xtb b/components/strings/components_chromium_strings_en-GB.xtb
index d10a1302d0..17bc0d4a87 100644
--- a/components/strings/components_chromium_strings_en-GB.xtb
+++ b/components/strings/components_chromium_strings_en-GB.xtb
@@ -5,7 +5,7 @@
 <translation id="275588974610408078">Crash reporting is not available in Chromium.</translation>
 <translation id="3064346599913645280">You're viewing a secure Chromium page</translation>
 <translation id="358997566136285270">Chromium logo</translation>
-<translation id="6411944759338344357">Cyfs Browser is made possible by the <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> open source project and other <ph name="BEGIN_LINK_OSS" />open source software<ph name="END_LINK_OSS" />.</translation>
+<translation id="8322499447224138136">Cyfs Browser is made possible by the <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> open source project and other <ph name="BEGIN_LINK_OSS" />open source software<ph name="END_LINK_OSS" />.</translation>
 <translation id="4559775032954821361">Go to
           the Chromium menu &gt;
           <ph name="SETTINGS_TITLE" />
diff --git a/components/strings/components_chromium_strings_zh-CN.xtb b/components/strings/components_chromium_strings_zh-CN.xtb
index 8a345070e8..6b3e13a1e7 100644
--- a/components/strings/components_chromium_strings_zh-CN.xtb
+++ b/components/strings/components_chromium_strings_zh-CN.xtb
@@ -5,7 +5,7 @@
 <translation id="275588974610408078">Chromium 未提供崩溃报告。</translation>
 <translation id="3064346599913645280">您正在查看安全的 Chromium 网页</translation>
 <translation id="358997566136285270">Chromium 徽标</translation>
-<translation id="6411944759338344357">Cyfs Browser 的诞生离不开 <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> 开放源代码项目以及其他<ph name="BEGIN_LINK_OSS" />开放源代码软件<ph name="END_LINK_OSS" />。</translation>
+<translation id="8322499447224138136">Cyfs Browser 的诞生离不开 <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> 开放源代码项目以及其他<ph name="BEGIN_LINK_OSS" />开放源代码软件<ph name="END_LINK_OSS" />。</translation>
 <translation id="4559775032954821361">依次转到 Chromium 菜单 &gt;“<ph name="SETTINGS_TITLE" />”&gt;“<ph name="ADVANCED_TITLE" />”&gt;“<ph name="PROXIES_TITLE" />”&gt;“LAN 设置”，然后取消选中“为 LAN 使用代理服务器”复选框。</translation>
 <translation id="4622039161600275920">此页面已被 Chromium 屏蔽</translation>
 <translation id="48558539577516920">在防火墙或防病毒设置部分设为允许 Chromium 访问网络。</translation>
