diff --git a/chrome/app/chromium_strings.grd b/chrome/app/chromium_strings.grd
index d8f79d505a..e381db8a38 100644
--- a/chrome/app/chromium_strings.grd
+++ b/chrome/app/chromium_strings.grd
@@ -378,7 +378,7 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
         <ph name="PAGE_TITLE">$1<ex>Google</ex></ph> - Chromium Canary
       </message>
       <message name="IDS_ABOUT_VERSION_COMPANY_NAME" desc="Company name on the about pages">
-        The Chromium Authors
+        The Cyfs Browser Authors
       </message>
       <message name="IDS_ABOUT_VERSION_COPYRIGHT" desc="Copyright information on the about pages">
         Copyright <ph name="YEAR">{0,date,y}<ex>2016</ex></ph> The Cyfs Browser Authors. All rights reserved.
@@ -606,7 +606,7 @@ Permissions you've already given to websites and apps may apply to this account.
         </message>
       </if>
       <message name="IDS_DEFAULT_BROWSER_INFOBAR_TEXT" desc="Text to show in an infobar when Chromium is not the current default browser.">
-        Chromium isn't your default browser
+        Cyfs Browser isn't your default browser
       </message>
       <message name="IDS_TAILORED_SECURITY_CONSENTED_ENABLE_NOTIFICATION_DESCRIPTION" desc="The text of the notification asking if the user wants to enable Enhanced Safe Browsing">
         You turned on Enhanced Safe Browsing in your account. Now get it for Chromium.
@@ -979,7 +979,7 @@ Permissions you've already given to websites and apps may apply to this account.
       <!-- Mac First-run dialog messages -->
       <if expr="is_macosx">
         <message name="IDS_FIRSTRUN_DLG_MAC_SET_DEFAULT_BROWSER_LABEL" desc="Label for checkbox that sets the default browser">
-          Set Chromium as your default browser
+          Set Cyfs Browser as your default browser
         </message>
       </if>
 
diff --git a/chrome/app/generated_resources.grd b/chrome/app/generated_resources.grd
index 37dd0fd704..030bdb52e9 100644
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -457,7 +457,7 @@ are declared in tools/grit/grit_rule.gni.
         Main menu
       </message>
       <message name="IDS_HISTORY_HISTORY_MENU_ITEM" desc="Label displayed in history sidebar button to display history.">
-        Chrome history
+        Cyfs Browser history
       </message>
       <message name="IDS_HISTORY_NO_SYNCED_RESULTS" desc="Text indicating that there are no synced tabs from other devices.">
         No tabs from other devices
diff --git a/chrome/app/resources/chromium_strings_en-GB.xtb b/chrome/app/resources/chromium_strings_en-GB.xtb
index 2bcf63df86..2d663e9d6d 100644
--- a/chrome/app/resources/chromium_strings_en-GB.xtb
+++ b/chrome/app/resources/chromium_strings_en-GB.xtb
@@ -176,7 +176,7 @@ Permissions that you've already given to websites and apps may apply to this acc
 <translation id="5386450000063123300">Updating Chromium (<ph name="PROGRESS_PERCENT" />)</translation>
 <translation id="538767207339317086">Allow Chromium sign-in</translation>
 <translation id="5398878173008909840">There is a new version of Chromium available.</translation>
-<translation id="5427571867875391349">Set Chromium as your default browser</translation>
+<translation id="2824236843868824522">Set Cyfs Browser as your default browser</translation>
 <translation id="5438241569118040789"><ph name="PAGE_TITLE" /> – Chromium Beta</translation>
 <translation id="5473971139929175403">Chromium may not function correctly because it is no longer supported on this Linux distribution</translation>
 <translation id="5480860683791598150">Chromium needs access to your location to share your location with this site</translation>
@@ -296,7 +296,7 @@ Google Chrome is unable to recover your settings.</translation>
 <translation id="7649070708921625228">Help</translation>
 <translation id="7682213815243802460">You can learn more about these features in Chromium settings.</translation>
 <translation id="7682601070171973634">Chromium is exploring new features that allow sites to deliver the same browsing experience using less of your data</translation>
-<translation id="7686590090926151193">Chromium isn't your default browser</translation>
+<translation id="4760585693357174315">Cyfs Browser isn't your default browser</translation>
 <translation id="7689606757190482937">Sync and personalise Chromium across your devices</translation>
 <translation id="7729447699958282447">Chromium could not sync your data because Sync is not available for your domain.</translation>
 <translation id="7745317241717453663">This will delete your browsing data from this device. To retrieve your data later, sign in to Chromium as <ph name="USER_EMAIL" />.</translation>
@@ -367,5 +367,5 @@ Google Chrome is unable to recover your settings.</translation>
 <translation id="945990405696787151">To get future Chromium updates, you'll need macOS 10.13 or later. This computer is using OS X 10.11.</translation>
 <translation id="965162752251293939">Who's using Chromium?</translation>
 <translation id="967427899662692980">Get Chromium's strongest security</translation>
-<translation id="985602178874221306">The Chromium Authors</translation>
+<translation id="2463748360542059515">The Cyfs Browser Authors</translation>
 </translationbundle>
\ No newline at end of file
diff --git a/chrome/app/resources/chromium_strings_zh-CN.xtb b/chrome/app/resources/chromium_strings_zh-CN.xtb
index 1d5f1c3081..14fdc837f8 100644
--- a/chrome/app/resources/chromium_strings_zh-CN.xtb
+++ b/chrome/app/resources/chromium_strings_zh-CN.xtb
@@ -174,7 +174,7 @@
 <translation id="5386450000063123300">正在更新 Chromium (<ph name="PROGRESS_PERCENT" />)</translation>
 <translation id="538767207339317086">允许登录 Chromium</translation>
 <translation id="5398878173008909840">Chromium 现已推出新版本。</translation>
-<translation id="5427571867875391349">将Chromium设为默认浏览器</translation>
+<translation id="5427571867875391349">将Cyfs Browser设为默认浏览器</translation>
 <translation id="5438241569118040789"><ph name="PAGE_TITLE" /> - Chromium 测试版</translation>
 <translation id="5480860683791598150">Chromium 需要获得位置权限，才能将您的位置信息共享给此网站</translation>
 <translation id="5487574057737591516">为了保护您的隐私，如果您的兴趣保留时间超过 4 周，系统便会自动将其删除。随着您继续浏览，某项兴趣可能会再次出现在列表中。如果 Chromium 的推测不准确，或者您不想看到某些广告，可以移除某项兴趣。</translation>
@@ -293,7 +293,7 @@ Chromium 无法恢复您的设置。</translation>
 <translation id="7649070708921625228">帮助</translation>
 <translation id="7682213815243802460">您可在 Chromium 设置中详细了解这些功能。</translation>
 <translation id="7682601070171973634">Chromium 团队正在探索新功能，以便网站只需使用您的较少数据就能提供同样的浏览体验</translation>
-<translation id="7686590090926151193">Chromium 不是您的默认浏览器</translation>
+<translation id="4760585693357174315">Cyfs Browser 不是您的默认浏览器</translation>
 <translation id="7689606757190482937">在您的所有设备上同步并个性化设置 Chromium</translation>
 <translation id="7729447699958282447">由于您的网域停用了同步功能，Chromium无法同步您的数据。</translation>
 <translation id="7745317241717453663">这会从这台设备上删除您的浏览数据。如果日后想要检索您的数据，届时请以 <ph name="USER_EMAIL" /> 的身份登录 Chromium。</translation>
@@ -363,5 +363,5 @@ Chromium 无法恢复您的设置。</translation>
 <translation id="945990405696787151">若要接收后续 Chromium 更新，您需使用 macOS 10.13 或更高版本。该计算机目前使用的是 OS X 10.11。</translation>
 <translation id="965162752251293939">谁在使用 Chromium？</translation>
 <translation id="967427899662692980">获享 Chromium 的最强安全保护</translation>
-<translation id="985602178874221306">The Chromium Authors</translation>
+<translation id="2463748360542059515">The Cyfs Browser Authors</translation>
 </translationbundle>
\ No newline at end of file
diff --git a/chrome/app/resources/generated_resources_en-GB.xtb b/chrome/app/resources/generated_resources_en-GB.xtb
index 3ac5176cce..78ec579fdd 100644
--- a/chrome/app/resources/generated_resources_en-GB.xtb
+++ b/chrome/app/resources/generated_resources_en-GB.xtb
@@ -1454,7 +1454,7 @@ You can manage this account’s settings by installing the Family Link app on yo
 <translation id="2336228925368920074">Bookmark All Tabs...</translation>
 <translation id="2336376423977300504">Always clear cookies when windows are closed</translation>
 <translation id="2336381494582898602">Powerwash</translation>
-<translation id="2337236196941929873">Preloads pages that Chrome thinks you might visit. To do this, Chrome may use cookies, if you allow cookies, and may encrypt and send pages through Google to hide your identity from sites.</translation>
+<translation id="7846298808536933440">Preloads pages that Cyfs Browser thinks you might visit. To do this, Cyfs Browser may use cookies, if you allow cookies, and may encrypt and send pages through Google to hide your identity from sites.</translation>
 <translation id="2340239562261172947"><ph name="FILE_NAME" /> can't be downloaded securely</translation>
 <translation id="2342180549977909852">Your child can use a number (PIN) instead of a password to unlock this device. To set a PIN later, go to Settings.</translation>
 <translation id="2342740338116612727">Bookmarks added</translation>
diff --git a/chrome/app/resources/generated_resources_zh-CN.xtb b/chrome/app/resources/generated_resources_zh-CN.xtb
index b351771ac5..78f017da4c 100644
--- a/chrome/app/resources/generated_resources_zh-CN.xtb
+++ b/chrome/app/resources/generated_resources_zh-CN.xtb
@@ -1432,7 +1432,7 @@
 <translation id="2336228925368920074">为所有标签页添加书签...</translation>
 <translation id="2336376423977300504">始终在关闭窗口时清除 Cookie</translation>
 <translation id="2336381494582898602">Powerwash</translation>
-<translation id="2337236196941929873">预加载 Chrome 认为您可能会访问的网页。为此，Chrome 可能会使用 Cookie（如果您允许使用 Cookie 的话），并通过 Google 加密和发送网页以在网站中隐藏您的身份。</translation>
+<translation id="7846298808536933440">预加载 Cyfs Browser 认为您可能会访问的网页。为此，Cyfs Browser 可能会使用 Cookie（如果您允许使用 Cookie 的话），并通过 Google 加密和发送网页以在网站中隐藏您的身份。</translation>
 <translation id="2340239562261172947">无法安全地下载“<ph name="FILE_NAME" />”</translation>
 <translation id="2342180549977909852">您的孩子可以使用数字（PIN 码）代替密码来解锁此设备。如果想等到需要时再设置 PIN 码，到时请前往“设置”。</translation>
 <translation id="2342740338116612727">书签已添加</translation>
@@ -5617,7 +5617,7 @@
 <translation id="652492607360843641">您已连接到<ph name="NETWORK_TYPE" />网络。</translation>
 <translation id="6527303717912515753">分享</translation>
 <translation id="6528179044667508675">请勿打扰</translation>
-<translation id="652948702951888897">Chrome 历史记录</translation>
+<translation id="6704509907164228464">Cyfs Browser 历史记录</translation>
 <translation id="6530186581263215931">这些设置是由您的管理员强制指定的</translation>
 <translation id="6531282281159901044">保留危险文件</translation>
 <translation id="6532101170117367231">保存到 Google 云端硬盘</translation>
diff --git a/chrome/app/settings_strings.grdp b/chrome/app/settings_strings.grdp
index 018492a176..6756d20a8e 100644
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1843,7 +1843,7 @@
     Uses cookies to remember your preferences, even if you don’t visit those pages
   </message>
   <message name="IDS_SETTINGS_NETWORK_PREDICTION_ENABLED_DESC_COOKIES_PAGE" desc="On the cookies page, the secondary text next to the checkbox that enables prediction of network actions.">
-    Preloads pages that Chrome thinks you might visit. To do this, Chrome may use cookies, if you allow cookies, and may encrypt and send pages through Google to hide your identity from sites.
+    Preloads pages that Cyfs Browser thinks you might visit. To do this, Cyfs Browser may use cookies, if you allow cookies, and may encrypt and send pages through Google to hide your identity from sites.
   </message>
   <message name="IDS_SETTINGS_SAFEBROWSING_ENABLEPROTECTION" desc="The section title of 'Protects you and your device from dangerous sites'">
     Safe Browsing (protects you and your device from dangerous sites)
diff --git a/chrome/browser/resources/cyfs_init/cyfs_init.html b/chrome/browser/resources/cyfs_init/cyfs_init.html
index 01b1e9bf29..75b3080979 100644
--- a/chrome/browser/resources/cyfs_init/cyfs_init.html
+++ b/chrome/browser/resources/cyfs_init/cyfs_init.html
@@ -3,7 +3,7 @@
 <head>
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
   <meta name="referrer" content="no-referrer">
-  <title i18n-content="welcomeMessage"></title>
+  <title i18n-content="welcomeMessage">Cyfs Browser is initializing</title>
   <link rel="stylesheet" href="cyfs_init.css">
   <script src="chrome://resources/js/load_time_data.js"></script>
   <script src="chrome://resources/js/util.js"></script>
diff --git a/chrome/browser/resources/history/side_bar.html b/chrome/browser/resources/history/side_bar.html
index ab2125c5fe..0358a7e428 100644
--- a/chrome/browser/resources/history/side_bar.html
+++ b/chrome/browser/resources/history/side_bar.html
@@ -120,13 +120,13 @@
           $i18n{historyMenuItem}
           <paper-ripple></paper-ripple>
         </a>
-        <a id="syncedTabs" role="menuitem" href="/syncedTabs"
+        <!-- <a id="syncedTabs" role="menuitem" href="/syncedTabs"
             class="page-item cr-nav-menu-item"
             path="syncedTabs" on-click="onItemClick_">
           <iron-icon icon="cr:phonelink"></iron-icon>
           $i18n{openTabsMenuItem}
           <paper-ripple></paper-ripple>
-        </a>
+        </a> -->
         <!-- tabindex is needed below because <a> without href attribute is not
              focusable in sequential keyboard navigation by default. -->
         <a role="menuitem" id="toggle-history-clusters"
diff --git a/chrome/browser/resources/settings/appearance_page/appearance_page.html b/chrome/browser/resources/settings/appearance_page/appearance_page.html
index a770f7b000..c06f234789 100644
--- a/chrome/browser/resources/settings/appearance_page/appearance_page.html
+++ b/chrome/browser/resources/settings/appearance_page/appearance_page.html
@@ -22,7 +22,7 @@
     <settings-animated-pages id="pages" section="appearance"
         focus-config="[[focusConfig_]]">
       <div route-path="default">
-        <div class="settings-row first" id="themeRow"
+        <!-- <div class="settings-row first" id="themeRow"
             hidden="[[!pageVisibility.setTheme]]">
           <cr-link-row class="first" hidden="[[!pageVisibility.setTheme]]"
               label="$i18n{themes}" sub-label="[[themeSublabel_]]"
@@ -60,7 +60,7 @@
             class="hr"
             hidden="[[!showHr_(
                 pageVisibility.setTheme, pageVisibility.homeButton)]]">
-        </div>
+        </div> -->
         <settings-toggle-button elide-label
             hidden="[[!pageVisibility.homeButton]]"
             pref="{{prefs.browser.show_home_button}}"
diff --git a/chrome/common/chrome_paths_mac.mm b/chrome/common/chrome_paths_mac.mm
index 41538c647a..abc2397ae1 100644
--- a/chrome/common/chrome_paths_mac.mm
+++ b/chrome/common/chrome_paths_mac.mm
@@ -59,7 +59,7 @@
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
       product_dir_name = "Google/Chrome";
 #else
-      product_dir_name = "Chromium";
+      product_dir_name = "cyber";
 #endif
     }
 
