diff --git a/build/win/reorder-imports.py b/build/win/reorder-imports.py
index 0995ae0058a006c22bd17e2c8199afd7caebebd2..11178cc9d64bb0d8eab0437eb204fc59dae5c733 100755
--- a/build/win/reorder-imports.py
+++ b/build/win/reorder-imports.py
@@ -25,8 +25,8 @@ def reorder_imports(input_dir, output_dir, architecture):
   # correct executable in the first place, so that this script
   # only needs to verify that and not write a whole new exe.
 
-  input_image = os.path.join(input_dir, 'chrome.exe')
-  output_image = os.path.join(output_dir, 'chrome.exe')
+  input_image = os.path.join(input_dir, 'CYFS_Browser.exe')
+  output_image = os.path.join(output_dir, 'CYFS_Browser.exe')
 
   # pefile mmap()s the whole executable, and then parses parts of
   # it into python data structures for ease of processing.
@@ -74,7 +74,7 @@ def reorder_imports(input_dir, output_dir, architecture):
 
   pe.write(filename=output_image)
 
-  for fname in glob.iglob(os.path.join(input_dir, 'chrome.exe.*')):
+  for fname in glob.iglob(os.path.join(input_dir, 'CYFS_Browser.exe.*')):
     shutil.copy(fname, os.path.join(output_dir, os.path.basename(fname)))
   return 0
 
