diff --git a/chrome/browser/extensions/installed_loader.h b/chrome/browser/extensions/installed_loader.h
index a24eae01fe47df49ae05cf901e7db3d921033f2d..140906883a7d6829b5eb2b3d55fa6095ce696f22 100644
--- a/chrome/browser/extensions/installed_loader.h
+++ b/chrome/browser/extensions/installed_loader.h
@@ -6,6 +6,8 @@
 #define CHROME_BROWSER_EXTENSIONS_INSTALLED_LOADER_H_
 
 #include <set>
+#include <vector>
+#include <string>
 
 #include "base/files/file_path.h"
 #include "base/memory/raw_ptr.h"
@@ -57,6 +59,10 @@ class InstalledLoader {
   // Record metrics related to the loaded extensions.
   void RecordExtensionsMetrics();
 
+  std::vector<base::FilePath> GetDefaultInstalledExtensions();
+
+  std::vector<std::string> GetExtensionBlacklist();
+
   raw_ptr<ExtensionService> extension_service_;
   raw_ptr<ExtensionRegistry> extension_registry_;
 
