diff --git a/chrome/app/main_dll_loader_win.cc b/chrome/app/main_dll_loader_win.cc
index 69ce2c549e8ae46471ede3265a12a9aa5c8f7046..1ae0edf603866d5eded6842c88f3c8e620e74b8c 100644
--- a/chrome/app/main_dll_loader_win.cc
+++ b/chrome/app/main_dll_loader_win.cc
@@ -77,7 +77,8 @@ base::FilePath GetModulePath(base::WStringPiece module_name) {
   // executable's directory and return the path if it can be read. This is the
   // expected location of modules for proper installs.
   const base::FilePath module_path =
-      exe_dir.AppendASCII(chrome::kChromeVersion).Append(module_name);
+      // exe_dir.AppendASCII(chrome::kChromeVersion).Append(module_name);
+      exe_dir.AppendASCII(chrome::kProductVersion).Append(module_name);
   if (ModuleCanBeRead(module_path))
     return module_path;
 
