diff --git a/chrome/installer/setup/uninstall.cc b/chrome/installer/setup/uninstall.cc
index d92390e2d73961c04d1e744818d98317bfb31cf6..fac50e04c5952159f34051ee0ca487a95e412f6b 100644
--- a/chrome/installer/setup/uninstall.cc
+++ b/chrome/installer/setup/uninstall.cc
@@ -365,9 +365,9 @@ InstallStatus IsChromeActiveOrUserCancelled(
   // - HUNG - chrome.exe was killed by HuntForZombieProcesses() (until we can
   //          give this method some brains and not kill chrome.exe launched
   //          by us, we will not uninstall if we get this return code).
-  VLOG(1) << "Launching Chrome to do uninstall tasks.";
+  VLOG(1) << "Launching CYFS browser to do uninstall tasks.";
   if (LaunchChromeAndWait(installer_state.target_path(), options, &exit_code)) {
-    VLOG(1) << "chrome.exe launched for uninstall confirmation returned: "
+    VLOG(1) << "CYFS_browser.exe launched for uninstall confirmation returned: "
             << exit_code;
     if ((exit_code == chrome::RESULT_CODE_UNINSTALL_CHROME_ALIVE) ||
         (exit_code == chrome::RESULT_CODE_UNINSTALL_USER_CANCEL) ||
@@ -377,7 +377,7 @@ InstallStatus IsChromeActiveOrUserCancelled(
     if (exit_code == chrome::RESULT_CODE_UNINSTALL_DELETE_PROFILE)
       return installer::UNINSTALL_DELETE_PROFILE;
   } else {
-    PLOG(ERROR) << "Failed to launch chrome.exe for uninstall confirmation.";
+    PLOG(ERROR) << "Failed to launch CYFS_browser.exe for uninstall confirmation.";
   }
 
   return installer::UNINSTALL_CONFIRMED;
