diff --git a/chrome/common/chrome_constants.cc b/chrome/common/chrome_constants.cc
index 6fec2c15bfc7cc73af63ae6d8844cbd414c79362..38346f77e6d4cf1cc359d46000ef9610a3acd002 100644
--- a/chrome/common/chrome_constants.cc
+++ b/chrome/common/chrome_constants.cc
@@ -118,6 +118,7 @@ const base::FilePath::CharType kExtensionsCookieFilename[] =
 const base::FilePath::CharType kFeatureEngagementTrackerStorageDirname[] =
     FPL("Feature Engagement Tracker");
 const base::FilePath::CharType kFirstRunSentinel[] = FPL("First Run");
+const base::FilePath::CharType kFirstRun[] = FPL("Cyfs First Run");
 const base::FilePath::CharType kGCMStoreDirname[] = FPL("GCM Store");
 const base::FilePath::CharType kLocalStateFilename[] = FPL("Local State");
 const base::FilePath::CharType kMediaCacheDirname[] = FPL("Media Cache");
