diff --git a/chrome/common/chrome_constants.cc b/chrome/common/chrome_constants.cc
index 6fec2c15bfc7cc73af63ae6d8844cbd414c79362..4ff4c0224f92017cba5b9fc78fe8c16c885afce5 100644
--- a/chrome/common/chrome_constants.cc
+++ b/chrome/common/chrome_constants.cc
@@ -15,7 +15,7 @@
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
 #define PRODUCT_STRING "Google Chrome"
 #elif BUILDFLAG(CHROMIUM_BRANDING)
-#define PRODUCT_STRING "Chromium"
+#define PRODUCT_STRING "CYFS Browser"
 #else
 #error Unknown branding
 #endif
@@ -118,6 +118,7 @@ const base::FilePath::CharType kExtensionsCookieFilename[] =
 const base::FilePath::CharType kFeatureEngagementTrackerStorageDirname[] =
     FPL("Feature Engagement Tracker");
 const base::FilePath::CharType kFirstRunSentinel[] = FPL("First Run");
+const base::FilePath::CharType kFirstRun[] = FPL("Cyfs First Run");
 const base::FilePath::CharType kGCMStoreDirname[] = FPL("GCM Store");
 const base::FilePath::CharType kLocalStateFilename[] = FPL("Local State");
 const base::FilePath::CharType kMediaCacheDirname[] = FPL("Media Cache");
