diff --git a/chrome/browser/extensions/installed_loader.h b/chrome/browser/extensions/installed_loader.h
index a24eae01fe47df49ae05cf901e7db3d921033f2d..c51acf6b1d6e6bdb3572fea25bf4c53fa2b8530b 100644
--- a/chrome/browser/extensions/installed_loader.h
+++ b/chrome/browser/extensions/installed_loader.h
@@ -6,9 +6,16 @@
 #define CHROME_BROWSER_EXTENSIONS_INSTALLED_LOADER_H_
 
 #include <set>
+#include <map>
 
 #include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/files/file_enumerator.h"
+#include "base/path_service.h"
+
 #include "base/memory/raw_ptr.h"
+#include "base/json/json_file_value_serializer.h"
+#include "base/json/json_string_value_serializer.h"
 
 namespace extensions {
 
@@ -45,6 +52,15 @@ class InstalledLoader {
   // Loads all installed extensions (used by startup and testing code).
   void LoadAllExtensions();
 
+  // Loads all default installed extensions.
+  void LoadDefaultExtensions();
+
+  void ReadDefaultExtensionPrefFile(base::DictionaryValue* prefs, base::FilePath json_file);
+
+  void UpdateDefaultExtensionPrefFile(const std::string& extension_id);
+
+  std::map<std::string, std::string> GetDefaultExtensionInfo();
+
   // Allows tests to verify metrics without needing to go through
   // LoadAllExtensions().
   void RecordExtensionsMetricsForTesting();
