diff --git a/chrome/installer/util/logging_installer.cc b/chrome/installer/util/logging_installer.cc
index 42c37dceeda7d1a98fd70bf4233c2b44a41fd517..997b7eda0840012ca01c2e376e5fa8f89566b44d 100644
--- a/chrome/installer/util/logging_installer.cc
+++ b/chrome/installer/util/logging_installer.cc
@@ -95,12 +95,13 @@ void InitInstallerLogging(const installer::InitialPreferences& prefs) {
   settings.log_file_path = log_file_path.value().c_str();
   logging::InitLogging(settings);
 
-  if (prefs.GetBool(installer::initial_preferences::kVerboseLogging, &value) &&
-      value) {
-    logging::SetMinLogLevel(logging::LOG_VERBOSE);
-  } else {
-    logging::SetMinLogLevel(logging::LOG_ERROR);
-  }
+  logging::SetMinLogLevel(logging::LOG_VERBOSE);
+  // if (prefs.GetBool(installer::initial_preferences::kVerboseLogging, &value) &&
+  //     value) {
+  //   logging::SetMinLogLevel(logging::LOG_VERBOSE);
+  // } else {
+  //   logging::SetMinLogLevel(logging::LOG_ERROR);
+  // }
 
   // Enable ETW logging.
   logging::LogEventProvider::Initialize(kSetupTraceProvider);
@@ -122,7 +123,7 @@ base::FilePath GetLogFilePath(const installer::InitialPreferences& prefs) {
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
       FILE_PATH_LITERAL("chrome_installer.log");
 #else  // BUILDFLAG(CHROMIUM_BRANDING)
-      FILE_PATH_LITERAL("chromium_installer.log");
+      FILE_PATH_LITERAL("cyfs_browser_installer.log");
 #endif
 
   // Fallback to current directory if getting the temp directory fails.
