diff --git a/chrome/browser/ui/browser.cc b/chrome/browser/ui/browser.cc
index 1837aac53e6e13aebb0eee00249b305e0c7192c9..902bb5c62e15045ea762a22ef6fe2e1cb8b38d0b 100644
--- a/chrome/browser/ui/browser.cc
+++ b/chrome/browser/ui/browser.cc
@@ -687,7 +687,7 @@ bool Browser::HasFindBarController() const {
 GURL Browser::GetNewTabURL() const {
   if (app_controller_)
     return app_controller_->GetAppStartUrl();
-  return GURL(chrome::kChromeUINewTabURL);
+  return GURL(chrome::kCyfsBrowserUINewTabURL);
 }
 
 gfx::Image Browser::GetCurrentPageIcon() const {
