diff --git a/chrome/app/theme/chromium/BRANDING b/chrome/app/theme/chromium/BRANDING
index f8363d5b294fe210ebfdd18374e95b54d6ff5c17..df62f5e4436c8053b7453833061c67178c652d76 100644
--- a/chrome/app/theme/chromium/BRANDING
+++ b/chrome/app/theme/chromium/BRANDING
@@ -1,10 +1,10 @@
-COMPANY_FULLNAME=The Chromium Authors
-COMPANY_SHORTNAME=The Chromium Authors
-PRODUCT_FULLNAME=Chromium
-PRODUCT_SHORTNAME=Chromium
-PRODUCT_INSTALLER_FULLNAME=Chromium Installer
-PRODUCT_INSTALLER_SHORTNAME=Chromium Installer
-COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Chromium Authors. All rights reserved.
-MAC_BUNDLE_ID=org.chromium.Chromium
+COMPANY_FULLNAME=CYFS Core Dev Team
+COMPANY_SHORTNAME=CYFS Core Dev Team
+PRODUCT_FULLNAME=CYFS Browser
+PRODUCT_SHORTNAME=CYFS Browser
+PRODUCT_INSTALLER_FULLNAME=CYFS Browser Installer
+PRODUCT_INSTALLER_SHORTNAME=CYFS Browser Installer
+COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The CYFS Core Dev Team. All rights reserved.
+MAC_BUNDLE_ID=com.cyfs.browser
 MAC_CREATOR_CODE=Cr24
 MAC_TEAM_ID=
