diff --git a/chrome/chrome_proxy/chrome_proxy_main_win.cc b/chrome/chrome_proxy/chrome_proxy_main_win.cc
index bdb02d140cf891e387d1f566e6f39692e0ebf35e..006384d1be0db460b661c032d1bb893a20773575 100644
--- a/chrome/chrome_proxy/chrome_proxy_main_win.cc
+++ b/chrome/chrome_proxy/chrome_proxy_main_win.cc
@@ -13,7 +13,7 @@
 namespace {
 
 constexpr base::FilePath::CharType kChromeExecutable[] =
-    FILE_PATH_LITERAL("chrome.exe");
+    FILE_PATH_LITERAL("CYFS_Browser.exe");
 
 constexpr base::FilePath::CharType kChromeProxyExecutable[] =
     FILE_PATH_LITERAL("chrome_proxy.exe");
