diff --git a/chrome/browser/extensions/extension_context_menu_model.h b/chrome/browser/extensions/extension_context_menu_model.h
index 8ed245724e2209bfe4a6782fc41070319b55370e..2c427bdc9369e2b078d0c518680970804fa93346 100644
--- a/chrome/browser/extensions/extension_context_menu_model.h
+++ b/chrome/browser/extensions/extension_context_menu_model.h
@@ -163,6 +163,9 @@ class ExtensionContextMenuModel : public ui::SimpleMenuModel,
   // Whether the menu is for a component extension.
   bool is_component_;
 
+  // Whether the extension can not uninstall.
+  bool is_cannot_uninstall_;
+
   // The extension action of the extension we are displaying the menu for (if
   // it has one, otherwise NULL).
   raw_ptr<ExtensionAction> extension_action_;
