diff --git a/chrome/browser/ui/toolbar/toolbar_actions_model.cc b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
index cc68dad50c994ffa7883d36ecc7cff5ac628eb6b..05e5618fae90bcb3c8d8b7cddbf5159c3bb69585 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
@@ -108,6 +108,15 @@ void ToolbarActionsModel::OnExtensionLoaded(
   // hides the browser action and then disables and enables the extension.
   if (!HasAction(extension->id()) && ShouldAddExtension(extension))
     AddAction(extension->id());
+
+  {
+    std::string extension_id = extension->id();
+    auto default_extension_ids = extension_prefs_->GetDefaultExtensions();
+    if (base::Contains(default_extension_ids, extension_id)) {
+      VLOG(1) << "Set extension " << extension_id << " pin";
+      SetActionVisibility(extension_id, true);
+    }
+  }
 }
 
 void ToolbarActionsModel::OnExtensionUnloaded(
