diff --git a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
index 7a63cdf64d3e0ccc1c10cf2cded3cec14843b24b..f45e9952fbc5dc28b9998a2d740ef9f429ed4f7e 100644
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -297,10 +297,11 @@ void AddAboutStrings(content::WebUIDataSource* html_source, Profile* profile) {
   std::string channel_name =
       chrome::GetChannelName(chrome::WithExtendedStable(true));
 #endif
-
+  std::string cyfs_browser_version = chrome::GetCYFSBrowserVersion();
+  std::string cyfs_version = cyfs_browser_version + std::string(" Chromium: ") + version_info::GetVersionNumber();
   std::u16string browser_version = l10n_util::GetStringFUTF16(
       IDS_SETTINGS_ABOUT_PAGE_BROWSER_VERSION,
-      base::UTF8ToUTF16(version_info::GetVersionNumber()),
+      base::UTF8ToUTF16(cyfs_version),
       l10n_util::GetStringUTF16(version_info::IsOfficialBuild()
                                     ? IDS_VERSION_UI_OFFICIAL
                                     : IDS_VERSION_UI_UNOFFICIAL),
@@ -316,6 +317,10 @@ void AddAboutStrings(content::WebUIDataSource* html_source, Profile* profile) {
 #endif
 
   html_source->AddString("aboutBrowserVersion", browser_version);
+  std::string runtime_version_string = chrome::GetCYFSRuntimeVersion();
+  std::string runtime_version = std::string("Module Version: ") + runtime_version_string;
+  html_source->AddString("aboutRuntimeVersion", runtime_version);
+
   html_source->AddString(
       "aboutProductCopyright",
       base::i18n::MessageFormatter::FormatWithNumberedArgs(
