diff --git a/chrome/common/channel_info_win.cc b/chrome/common/channel_info_win.cc
index f2731333dc397591cd995ea60eb823822f6ca2b1..30c8dc787258281bedf91d70dbd332d269f4604f 100644
--- a/chrome/common/channel_info_win.cc
+++ b/chrome/common/channel_info_win.cc
@@ -4,12 +4,22 @@
 
 #include "chrome/common/channel_info.h"
 
+#include "base/logging.h"
+#include "base/files/file_path.h"
 #include "base/debug/profiler.h"
 #include "base/strings/string_util.h"
 #include "base/strings/utf_string_conversions.h"
 #include "build/branding_buildflags.h"
 #include "chrome/install_static/install_util.h"
 
+#include "base/files/file_util.h"
+#include "base/path_service.h"
+
+#include <fstream>
+#include <shlobj.h>
+#include <wrl.h>
+#include <Windows.h>
+
 namespace chrome {
 
 std::string GetChannelName(WithExtendedStable with_extended_stable) {
@@ -27,6 +37,61 @@ std::string GetChannelName(WithExtendedStable with_extended_stable) {
 #endif
 }
 
+std::string GetCYFSRuntimeVersion() {
+  std::string version_str;
+  base::FilePath user_data_path;
+  if (!base::PathService::Get(base::DIR_ROAMING_APP_DATA, &user_data_path))
+    return version_str;
+
+  user_data_path = user_data_path.AppendASCII("cyfs")
+    .AppendASCII("services")
+    .AppendASCII("runtime")
+    .AppendASCII("www")
+    .AppendASCII("version.html");
+
+  std::ifstream input_file(user_data_path.AsUTF8Unsafe(), std::ios::binary);
+  if (!input_file.is_open()) {
+	  LOG(INFO) << "failed to open " << user_data_path;
+	  return version_str;
+  }
+
+  while (input_file) {
+    std::string input_buffer(1024, '\0');
+    input_file.read(&input_buffer.front(), input_buffer.size());
+    version_str += input_buffer.substr(0, input_file.gcount());
+  }
+  LOG(INFO) << "Cyfs runtime version " << version_str;
+  return version_str;
+}
+
+std::string GetCYFSBrowserVersion() {
+  std::string version_str;
+  base::FilePath user_data_path;
+  if (!base::PathService::Get(base::DIR_ROAMING_APP_DATA, &user_data_path))
+    return version_str;
+
+  user_data_path = user_data_path.AppendASCII("cyfs")
+    .AppendASCII("services")
+    .AppendASCII("runtime")
+    .AppendASCII("www")
+    .AppendASCII("browser_version.html");
+
+  std::ifstream input_file(user_data_path.AsUTF8Unsafe(), std::ios::binary);
+  if (!input_file.is_open()) {
+	  LOG(INFO) << "failed to open " << user_data_path;
+	  return version_str;
+  }
+
+  while (input_file) {
+    std::string input_buffer(1024, '\0');
+    input_file.read(&input_buffer.front(), input_buffer.size());
+    version_str += input_buffer.substr(0, input_file.gcount());
+  }
+  LOG(INFO) << "Cyfs browser version " << version_str;
+  return version_str;
+}
+
+
 version_info::Channel GetChannel() {
   return install_static::GetChromeChannel();
 }
