diff --git a/components/os_crypt/keychain_password_mac.mm b/components/os_crypt/keychain_password_mac.mm
index 5589310e2e1f41a6a97e77bb57a7a71cd09a18be..e58a7e98203c7c681fa5a7c445b83e12a4064cc6 100644
--- a/components/os_crypt/keychain_password_mac.mm
+++ b/components/os_crypt/keychain_password_mac.mm
@@ -31,8 +31,8 @@ namespace {
 const char kDefaultServiceName[] = "Chrome Safe Storage";
 const char kDefaultAccountName[] = "Chrome";
 #else
-const char kDefaultServiceName[] = "Chromium Safe Storage";
-const char kDefaultAccountName[] = "Chromium";
+const char kDefaultServiceName[] = "CYFS Browser Safe Storage";
+const char kDefaultAccountName[] = "CYFS Browser";
 #endif
 
 // Generates a random password and adds it to the Keychain.  The added password
