diff --git a/content/browser/loader/navigation_url_loader_impl.cc b/content/browser/loader/navigation_url_loader_impl.cc
index dbecaf0a549233e22acd91d231e7d3e11ad6e4d7..34d0e4bba123279c85bfcc34cbb290644680aa39 100644
--- a/content/browser/loader/navigation_url_loader_impl.cc
+++ b/content/browser/loader/navigation_url_loader_impl.cc
@@ -529,6 +529,7 @@ void NavigationURLLoaderImpl::MaybeStartLoader(
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   DCHECK(started_);
 
+  ReplaceRequestUrl();
   if (single_request_factory) {
     // `interceptor` wants to handle the request with
     // `single_request_handler`.
@@ -622,6 +623,26 @@ void NavigationURLLoaderImpl::MaybeStartLoader(
       GetUIThreadTaskRunner({BrowserTaskType::kNavigationNetworkResponse}));
 }
 
+void NavigationURLLoaderImpl::ReplaceRequestUrl() {
+  GURL request_url = resource_request_->url;
+  if (request_url.SchemeIsCYFS()) {
+    // translate cyfs url to local http url
+    auto host_str = request_url.host();
+    auto old_path_str = request_url.path();
+    auto new_url_path_str = "/" + host_str + old_path_str;
+
+    GURL local_cyfs_gateway_url("http://127.0.0.1:38090");
+    GURL::Replacements replacements;
+    replacements.SetSchemeStr(local_cyfs_gateway_url.scheme_piece());
+    replacements.SetHostStr(local_cyfs_gateway_url.host_piece());
+    replacements.SetPortStr(local_cyfs_gateway_url.port_piece());
+    replacements.SetPathStr(new_url_path_str);
+    GURL new_url = request_url.ReplaceComponents(replacements);
+    LOG(INFO) << "Update request url " << request_url.spec() << " to " << new_url.spec();
+    resource_request_->url = new_url;
+  }
+}
+
 void NavigationURLLoaderImpl::FallbackToNonInterceptedRequest(
     bool reset_subresource_loader_params) {
   if (reset_subresource_loader_params)
