diff --git a/extensions/browser/pref_names.h b/extensions/browser/pref_names.h
index 2ef8a2d2bd16f1e9e3b20d32ed40b029cb478073..a29173f1d767ac63e8259dff265870da53cae177 100644
--- a/extensions/browser/pref_names.h
+++ b/extensions/browser/pref_names.h
@@ -134,6 +134,8 @@ extern const char kPrefContentSettings[];
 // Extension-set incognito content settings.
 extern const char kPrefIncognitoContentSettings[];
 
+extern const char KDefaultExtensionIdList[];
+
 }  // namespace pref_names
 }  // namespace extensions
 
