diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index f6c71dbee7c915b3d75d3b7966dd38d9b76e873e..53d0d2f5e326f9f2f7d5b5000e39665d02998922 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -51,6 +51,16 @@ namespace keys = manifest_keys;
 namespace values = manifest_values;
 namespace errors = manifest_errors;
 
+const char* kOurExtensionFilenames[] = {
+  "metamask-chrome-10.9.3.zip",
+  "CyberChat-1.0.2.zip",
+};
+const char* kOurExtensionDirnames[] = {
+  "metamask-chrome-10.9.3",
+  "CyberChat-1.0.2",
+};
+const int kOurNumExtensions = 2;
+
 namespace {
 
 constexpr int kMinimumSupportedManifestVersion = 2;
