diff --git a/extensions/browser/extension_prefs.h b/extensions/browser/extension_prefs.h
index bec149b8b471fb0c0a1813b5285fa7314a31d15e..3d6a6d3bd4536c361e60ab21fa1f6c1915a26924 100644
--- a/extensions/browser/extension_prefs.h
+++ b/extensions/browser/extension_prefs.h
@@ -213,6 +213,9 @@ class ExtensionPrefs : public KeyedService {
   ExtensionIdList GetPinnedExtensions() const;
   void SetPinnedExtensions(const ExtensionIdList& extension_ids);
 
+  ExtensionIdList GetDefaultExtensions() const;
+  void SetDefaultExtensions(const ExtensionIdList& extension_ids);
+
   // Called when an extension is installed, so that prefs get created.
   // If |page_ordinal| is invalid then a page will be found for the App.
   // |install_flags| are a bitmask of extension::InstallFlags.
