diff --git a/net/cookies/cookie_monster.cc b/net/cookies/cookie_monster.cc
index 5f54517f7a6905f6c4ba026c620bbb4c15f4efc3..83388949881bfe6ea1cd9ecdee81ec3d4b2ea5ba 100644
--- a/net/cookies/cookie_monster.cc
+++ b/net/cookies/cookie_monster.cc
@@ -557,8 +557,8 @@ void CookieMonster::SetPersistSessionCookies(bool persist_session_cookies) {
   persist_session_cookies_ = persist_session_cookies;
 }
 
-const char* const CookieMonster::kDefaultCookieableSchemes[] = {"http", "https",
-                                                                "ws", "wss"};
+const char* const CookieMonster::kDefaultCookieableSchemes[] = {
+    "http", "https", "ws", "wss", "cyfs", "ipfs", "ipns"};
 const int CookieMonster::kDefaultCookieableSchemesCount =
     std::size(kDefaultCookieableSchemes);
 
@@ -613,8 +613,9 @@ void CookieMonster::GetAllCookies(GetAllCookiesCallback callback) {
 
   CookieList cookie_list;
   cookie_list.reserve(cookie_ptrs.size());
-  for (auto* cookie_ptr : cookie_ptrs)
+  for (auto* cookie_ptr : cookie_ptrs) {
     cookie_list.push_back(*cookie_ptr);
+  }
 
   MaybeRunCookieCallback(std::move(callback), cookie_list);
 }
