diff --git a/net/url_request/url_request.h b/net/url_request/url_request.h
index 4a4d0a87babc9b14bb5d29bce3f511c21e0fe2d2..224d38c34a8def22cc5770d9cf2f8e90789c8f3b 100644
--- a/net/url_request/url_request.h
+++ b/net/url_request/url_request.h
@@ -226,6 +226,8 @@ class NET_EXPORT URLRequest : public base::SupportsUserData {
   // started. Once it was set to block all cookies, it cannot be changed back.
   static void SetDefaultCookiePolicyToBlock();
 
+  void PushURL(const GURL& url);
+
   // The original url is the url used to initialize the request, and it may
   // differ from the url if the request was redirected.
   const GURL& original_url() const { return url_chain_.front(); }
