diff --git a/services/network/public/cpp/url_util.cc b/services/network/public/cpp/url_util.cc
index f41b8934ff7bc17f5d2f0cc4242bb83fd8ac8704..1fc7ffd60fa9bd0969b290543544f482fda7f4ee 100644
--- a/services/network/public/cpp/url_util.cc
+++ b/services/network/public/cpp/url_util.cc
@@ -6,10 +6,18 @@
 
 #include "url/gurl.h"
 
+namespace {
+
+bool IsURLHandledByCYFSNetworkService(const GURL& url) {
+  return (url.SchemeIsCYFS() || url.SchemeIsIPFS() || url.SchemeIsIPNS());
+}
+
+}
+
 namespace network {
 
 bool IsURLHandledByNetworkService(const GURL& url) {
-  return (url.SchemeIsHTTPOrHTTPS() || url.SchemeIsWSOrWSS());
+  return (url.SchemeIsHTTPOrHTTPS() || url.SchemeIsWSOrWSS() || IsURLHandledByCYFSNetworkService(url));
 }
 
 }  // namespace network
