diff --git a/third_party/blink/renderer/platform/wtf/text/string_statics.cc b/third_party/blink/renderer/platform/wtf/text/string_statics.cc
index aeb0d1d2d0a96672ecd63b94eda8b3e19ae5cc49..1a7b3ca97b798ef45a01de5427639574dd4ba7c9 100644
--- a/third_party/blink/renderer/platform/wtf/text/string_statics.cc
+++ b/third_party/blink/renderer/platform/wtf/text/string_statics.cc
@@ -40,6 +40,8 @@ WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_xml_atom);
 WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_xmlns_atom);
 WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_xlink_atom);
 WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_http_atom);
+WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_cyfs_atom);
+WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_ipfs_atom);
 WTF_EXPORT DEFINE_GLOBAL(AtomicString, g_https_atom);
 
 // This is not an AtomicString because it is unlikely to be used as an
@@ -93,6 +95,10 @@ void StringStatics::Init() {
   new (NotNullTag::kNotNull, (void*)&g_xmlns_with_colon) String("xmlns:");
   new (NotNullTag::kNotNull, (void*)&g_http_atom)
       AtomicString(AddStaticASCIILiteral("http"));
+  new (NotNullTag::kNotNull, (void*)&g_cyfs_atom)
+      AtomicString(AddStaticASCIILiteral("cyfs"));
+  new (NotNullTag::kNotNull, (void*)&g_ipfs_atom)
+      AtomicString(AddStaticASCIILiteral("ipfs"));
   new (NotNullTag::kNotNull, (void*)&g_https_atom)
       AtomicString(AddStaticASCIILiteral("https"));
 }
