diff --git a/third_party/blink/renderer/platform/wtf/text/atomic_string.h b/third_party/blink/renderer/platform/wtf/text/atomic_string.h
index 064d53f0a0399d0bffa3ef0c5af4f22a0fb52ce7..27de801a003520d0c6f759971a25bac8265fdae7 100644
--- a/third_party/blink/renderer/platform/wtf/text/atomic_string.h
+++ b/third_party/blink/renderer/platform/wtf/text/atomic_string.h
@@ -287,6 +287,8 @@ WTF_EXPORT extern const AtomicString& g_star_atom;
 WTF_EXPORT extern const AtomicString& g_xml_atom;
 WTF_EXPORT extern const AtomicString& g_xmlns_atom;
 WTF_EXPORT extern const AtomicString& g_xlink_atom;
+WTF_EXPORT extern const AtomicString& g_cyfs_atom;
+WTF_EXPORT extern const AtomicString& g_ipfs_atom;
 WTF_EXPORT extern const AtomicString& g_http_atom;
 WTF_EXPORT extern const AtomicString& g_https_atom;
 
