diff --git a/url/gurl.h b/url/gurl.h
index 8f4884c7a59f8fdae0f24c410b55a55fe701c029..76a785afc112c657537878ec75d594c78b7182a9 100644
--- a/url/gurl.h
+++ b/url/gurl.h
@@ -232,6 +232,15 @@ class COMPONENT_EXPORT(URL) GURL {
   // colon.
   bool SchemeIs(base::StringPiece lower_ascii_scheme) const;
 
+  // Returns true if the scheme is "cyfs".
+  bool SchemeIsCYFS() const;
+
+  // Returns true if the scheme is "ipfs".
+  bool SchemeIsIPFS() const;
+
+  // Returns true if the scheme is "ipns".
+  bool SchemeIsIPNS() const;
+
   // Returns true if the scheme is "http" or "https".
   bool SchemeIsHTTPOrHTTPS() const;
 
