diff --git a/url/url_constants.cc b/url/url_constants.cc
index 96850982c936e6d4cfd536f3c5dcff8ebfdabefa..7513aac5cd41857f0ab534418e8c224f1296de14 100644
--- a/url/url_constants.cc
+++ b/url/url_constants.cc
@@ -32,6 +32,12 @@ const char kFileSystemScheme[] = "filesystem";
 const char16_t kFileSystemScheme16[] = u"filesystem";
 const char kFtpScheme[] = "ftp";
 const char16_t kFtpScheme16[] = u"ftp";
+const char kCyfsScheme[] = "cyfs";
+const char16_t kCyfsScheme16[] = u"cyfs";
+extern const char kIpfsScheme[] = "ipfs";
+extern const char16_t kIpfsScheme16[] = u"ipfs";
+extern const char kIpnsScheme[] = "ipns";
+extern const char16_t kIpnsScheme16[] = u"ipns";
 const char kHttpScheme[] = "http";
 const char16_t kHttpScheme16[] = u"http";
 const char kHttpsScheme[] = "https";
