diff --git a/url/gurl.cc b/url/gurl.cc
index e01364690efe15e0cc2ca7e3e33ec0eadd41aa3a..adfee77d6c7d095820eb79cde91435eaee8dd5f8 100644
--- a/url/gurl.cc
+++ b/url/gurl.cc
@@ -352,6 +352,9 @@ bool GURL::SchemeIs(base::StringPiece lower_ascii_scheme) const {
   return scheme_piece() == lower_ascii_scheme;
 }
 
+bool GURL::SchemeIsCYFS() const {
+  return SchemeIs(url::kCyfsScheme); }
+
 bool GURL::SchemeIsHTTPOrHTTPS() const {
   return SchemeIs(url::kHttpScheme) || SchemeIs(url::kHttpsScheme);
 }
