diff --git a/front_end/panels/application/ApplicationPanelSidebar.ts b/front_end/panels/application/ApplicationPanelSidebar.ts
index c03c995e8d4dc248e19d9f4a3684cd1c81a95b1a..792c0521a7de50249d14e9ca3a4ea7519f05c3e3 100644
--- a/front_end/panels/application/ApplicationPanelSidebar.ts
+++ b/front_end/panels/application/ApplicationPanelSidebar.ts
@@ -579,7 +579,7 @@ export class ApplicationPanelSidebar extends UI.Widget.VBox implements SDK.Targe
     // debug why the frame was unreachable.
     const urlToParse = frame.unreachableUrl() || frame.url;
     const parsedURL = Common.ParsedURL.ParsedURL.fromString(urlToParse);
-    if (!parsedURL || (parsedURL.scheme !== 'http' && parsedURL.scheme !== 'https' && parsedURL.scheme !== 'file')) {
+    if (!parsedURL || (parsedURL.scheme !== 'cyfs' && parsedURL.scheme !== 'http' && parsedURL.scheme !== 'https' && parsedURL.scheme !== 'file')) {
       return;
     }
 
